### DISCUSSION TITLE
Newb help: isorecursive vs nominal modes

### DISCUSSION THREAD

**Discussion by trusktr** [WebAssembly/binaryen#5788] [Q&A]
Is the main difference kinda similar to statically linked builds vs dynamically linked? For example, I am imagining isorecursive mode during optimization not eliminating things means they can be unused, but will be used later when dynamically linked. Am I thinking correctly? Or do the two modes serve totally different purposes that what I just imagined?

**Comment by kripken** [comment]
Nominal mode was mainly useful before the Wasm GC spec was finalized. It was an option for how the spec could have ended up. The final spec has elements of nominal and structural typing.

Recent binaryen versions remove nominal mode, so it's not relevant any more.

Yes, this is related to dynamic linking, in a sense: Structural typing is useful when you want to link two modules together, and want types to match up. Nominal typing is useful inside a module to differentiate things that are similar structurally but have different meanings. Both are possible in the final wasm GC type system.
